core/src/main/java/org/opensearch/sql/expression/function/RelevanceFunctionResolver.java

```diff
@@ -0,0 +1,52 @@
+package org.opensearch.sql.expression.function;
+
+import java.util.List;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.tuple.Pair;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.exception.SemanticCheckException;
+
+@RequiredArgsConstructor
+public class RelevanceFunctionResolver
+    implements FunctionResolver {
+
+  @Getter
+  private final FunctionName functionName;
+
+  @Getter
+  private final ExprType declaredFirstParamType;
+
+  @Override
+  public Pair<FunctionSignature, FunctionBuilder> resolve(FunctionSignature unresolvedSignature) {
+    if (!unresolvedSignature.getFunctionName().equals(functionName)) {
+      throw new SemanticCheckException(String.format("Expected '%s' but got '%s'",
+          functionName.getFunctionName(), unresolvedSignature.getFunctionName().getFunctionName()));
+    }
+    List<ExprType> paramTypes = unresolvedSignature.getParamTypeList();
+    ExprType providedFirstParamType = paramTypes.get(0);
+
+    if (!declaredFirstParamType.equals(providedFirstParamType)) {
+      throw new SemanticCheckException(
+          getWrongParameterErrorMessage(0, providedFirstParamType, declaredFirstParamType));
+    }
+
+    for (int i = 1; i < paramTypes.size(); i++) {
```

Comment:
Maybe a little comment might be nice to avoid someone having to code trace in order to understand why the use of the magic number.

Reply:
Addressed in https://github.com/Bit-Quill/opensearch-project-sql/pull/99/commits/277ee6ad9485d4a0ad3097cec7af916894e14284